### DISCUSSION TITLE
N8N and Windmill integrations

### DISCUSSION THREAD

**Discussion by pombredanne** [twentyhq/twenty#3828] [Ideas]
Zapier is fine, but not great for an open source tool. Have you considered other integration brokers/platforms ?

**Comment by pombredanne** [comment]
May be something like https://github.com/huginn/huginn ?

**Reply by FelixMalfait** [reply]
https://github.com/windmill-labs/windmill is a great option, we need to work on an "official" integration but it's relatively easy to setup within Windmill directly.

We also need to support n8n long-term which is the most popular "open source Zapier" but nobody's working on that at the moment

**Comment by tempusthales** [comment]
n8n integration would be really great, along with windmill which I am a big fan of.

**Comment by thrmda** [comment]
n8n Please

**Reply by urinka** [reply]
n8n + 1

**Comment by JakubHelis** [comment]
n8n +1

**Comment by luiszxcv** [comment]
N8N is necessary in here

**Comment by boringcommerce** [comment]
n8n & activepieces pls

**Comment by tradaonline** [comment]
n8n please

**Comment by clarkbalan** [comment]
n8n please

**Comment by shodgson** [comment]
I have started an [n8n-nodes-twenty integration](https://www.npmjs.com/package/n8n-nodes-twenty) available as an community node inside n8n.
At the moment it supports all the standard operations in Twenty v0.24.1, however it is still an early release and likely to break.

Edit: now compatible with version v0.40

**Reply by shodgson** [reply]
The API is currently broken in Twenty v0.32.0 as reported in issue [#8925](https://github.com/twentyhq/twenty/issues/8295).  After it is fixed, I will be able to update the integration to include the latest endpoints.

**Reply by shodgson** [reply]
:heavy_check_mark:  The n8n integration is working for v0.33.4

**Reply by du8die** [reply]
@shodgson - Good Plugin. If you could find a way to include custom fields, it would be AMAZING!

**Reply by shodgson** [reply]
Thanks @du8die. The API is not stable enough yet for me to add more features right now - but I've added an [issue](https://github.com/shodgson/n8n-nodes-twenty/issues/1) to keep track of your request.

**Comment by vincenzodomina** [comment]
Definitely n8n, please!

**Comment by BOHEUS** [comment]
For all interested, Twenty will implement its' own version of n8n called workflows which first version should be available by the end of this year (source: #7651) so there won't be a need to use third-party tool to implement your workflows.

Also, shodgson's repo is outdated as new endpoints are missing

**Reply by parion** [reply]
Appears work is underway! 😄 
https://github.com/twentyhq/twenty/issues/8721

**Reply by FelixMalfait** [reply]
It is :) - basic version is working but still work to do to make it a great user experience!

<img width="1200" alt="Screenshot 2024-11-27 at 10 55 06" src="https://github.com/user-attachments/assets/bc4fd274-a931-4344-9d34-934310f7e581">

**Reply by shodgson** [reply]
This looks nice.
However, I would prefer to see more bug fixes[1] in Twenty instead of introducing new features like this. n8n and Windmill do workflows _really_ well.  It would be easier to develop plugins and extensions if the API and product in general is more stable.

P.S. Felix & co, I think you guys are doing an amazing job developing a cool project, and thanks for all the work you have contributed to the open source community. I am sure you are getting lots of input/pressure from customer, investors, developers, etc, so I trust you will make the right call for Twenty overall. Above is just my 2 cents.

1. E.g., I think the API is broken again when self-hosting v0.33.4.

**Reply by shodgson** [reply]
And the n8n community node (n8n-nodes-twenty) is now working for v0.33.4.

**Reply by FelixMalfait** [reply]
Hey @shodgson thanks for the feedback, it's a though balance to find! We wanted to keep a lot of freedom to iterate in 2024. 2025 will be the year Twenty becomes much more stable (we'll have less and less big changes to the underlying data model structure as we finally got most things right there, etc.)

**Reply by raphspeaks** [reply]
> This looks nice. However, I would prefer to see more bug fixes[1] in Twenty instead of introducing new features like this. n8n and Windmill do workflows _really_ well. It would be easier to develop plugins and extensions if the API and product in general is more stable.
> 
> P.S. Felix & co, I think you guys are doing an amazing job developing a cool project, and thanks for all the work you have contributed to the open source community. I am sure you are getting lots of input/pressure from customer, investors, developers, etc, so I trust you will make the right call for Twenty overall. Above is just my 2 cents.
> 
> 1. E.g., I think the API is broken again when self-hosting v0.33.4.

It 100% is, and I posted an issue on your repo. The adjustments are pretty straightforward. I would probably just focus on a single node and then work through the rest as a priority. 

@BOHEUS do you know much about what will be included in the workflows as it relates to syncing data to other external apps?

**Reply by BOHEUS** [reply]
@raphspeaks sadly I don't have such information

**Reply by FelixMalfait** [reply]
@raphspeaks By early April Workflows will be released publicly, meaning it's stable and supports all the basic tasks (branching, delay, cron trigger, code node, etc.). We plan a public alpha release end of Jan (meaning you will be able to toggle it in an 'early access' page in Settings).

@shodgson's integration is still very useful because n8n has a rich ecosystem of apps. Later we might work on integrating external apps through pieces from ActivePieces.com directly into Twenty (we can't integrate n8n nodes because their license doesn't allow it), but I'm not sure it's relevant, as @shodgson pointed out - top priority should be to reinforce the current scope.

**Comment by francescogruner** [comment]
N8N +1

**Comment by Bonapara** [comment]
We are working on our own workflow engine (available in the Labs page in Settings).
